torch/_dynamo/variables/torch.py

```diff
@@ -148,36 +148,45 @@ def _dynamo_overriden_transformers_eq(self, other):
     pass
 
 
-def torch_reconstruct(codegen, value):
```

Comment:
How should I understand, at a high level, the changes in this file?

Reply:
I refactor out a ```BaseTorchVariable``` as the base class for all torch variables. 
* ```TorchCtxManagerClassVariable```: torch context manager classes implemented by Dynamo;
* ```TorchInGraphFunctionVariable```: in graph functions (including constant folding functions and Dynamo implemented functions)
* ```TorchVariable```: The rest ```is_allowed``` torch classes/objects/functions. I'll further refactor this, and probably I'll remove this variable to in favor of ```UserDefineClassVariable```, ```PythonModuleVariable```, ```UserFunctionVariables``` for inlining ones, ```SkipFilesVariable``` for skipped ones.